service/src/main/java/com/epam/esm/service/validator/CertificateDtoValidator.java

```diff
@@ -0,0 +1,138 @@
+package com.epam.esm.service.validator;
+
+import com.epam.esm.service.constant.CustomHttpStatus;
+import com.epam.esm.service.dto.CertificateDto;
+import com.epam.esm.service.dto.ErrorMessageDto;
+import com.epam.esm.service.locale.LanguageManager;
+import com.epam.esm.service.locale.LocaleMessages;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+@Component
+public class CertificateDtoValidator {
+  private static final String NAME_REGEX = "[a-zA-Z][a-zA-Z0-9_]{2,30}";
+  private static final String DESCRIPTION_REGEX = "[a-zA-Z0-9_,.\\/\\\\<> ;:?!-]{2,300}";
+
+  private final TagDtoValidator tagDtoValidator;
+
+  @Autowired
+  public CertificateDtoValidator(TagDtoValidator tagDtoValidator) {
+    this.tagDtoValidator = tagDtoValidator;
+  }
+
+  public List<ErrorMessageDto> createCertificateDtoValidate(CertificateDto certificateDto) {
```

Comment:
@Exqzore Looks scary :) I think it can be simplier.

Reply:
fixed